releng/release-tool/release-tool.go

```diff
@@ -161,40 +176,56 @@ func (r *releaseData) generateReleaseNotes() error {
 	numContributors := len(contributorList)
 	typeOfChanges = strings.TrimSpace(typeOfChanges)
 
-	f.WriteString(fmt.Sprintf("This release follows %s and consists of %d changes, contributed by %d people, leading to %s.\n", r.previousTag, numChanges, numContributors, typeOfChanges))
+	f.WriteString(fmt.Sprintf("# %s %s\n", r.name, r.tag))
+	f.WriteString("\n")
+	f.WriteString(fmt.Sprintf("This release follows %s and consists of %d changes, leading to %s.", r.previousTag, numChanges, typeOfChanges))
 	if r.promoteRC != "" {
-		f.WriteString(fmt.Sprintf("%s is a promotion of release candidate %s which was originally published %s", r.tag, r.promoteRC, r.promoteRCTime.Format("2006-01-02")))
+		f.WriteString("  \n")
+		f.WriteString(fmt.Sprintf("%s is a promotion of release candidate %s, which was originally published on %s.", r.tag, r.promoteRC, r.promoteRCTime.Format("2006-01-02")))
 	}
 	f.WriteString("\n")
-	f.WriteString(fmt.Sprintf("The source code and selected binaries are available for download at: %s.\n", tagUrl))
 	f.WriteString("\n")
-	f.WriteString("The primary release artifact of KubeVirt is the git tree. The release tag is\n")
+	f.WriteString(fmt.Sprintf("The primary release artifact of %s is the git tree. The release tag is ", r.name))
 	f.WriteString(fmt.Sprintf("signed and can be verified using `git tag -v %s`.\n", r.tag))
 	f.WriteString("\n")
-	f.WriteString(fmt.Sprintf("Pre-built containers are published on Quay and can be viewed at: <https://quay.io/%s/>.\n", r.org))
+	f.WriteString(fmt.Sprintf("The source code and selected binaries are available for download at: %s  \n", tagUrl))
+	f.WriteString(fmt.Sprintf("Pre-built containers are published on Quay and can be viewed at: https://quay.io/%s/\n", r.org))
 	f.WriteString("\n")
 
 	if len(releaseNotes) > 0 {
-		f.WriteString("Notable changes\n---------------\n")
+		f.WriteString("## Notable changes\n")
 		f.WriteString("\n")
 		for _, note := range releaseNotes {
 			f.WriteString(fmt.Sprintf("- %s\n", note))
 		}
 	}
 
 	f.WriteString("\n")
-	f.WriteString("Contributors\n------------\n")
-	f.WriteString(fmt.Sprintf("%d people contributed to this release:\n\n", numContributors))
+	f.WriteString("## Contributors\n")
+	f.WriteString("\n")
+	f.WriteString(fmt.Sprintf("%d people contributed to this release:\n", numContributors))
+	f.WriteString("\n")
 
+	f.WriteString("```\n")
 	for _, contributor := range contributorList {
-		if strings.Contains(contributor, "kubevirt-bot") {
-			// skip the bot
-			continue
-		}
 		f.WriteString(fmt.Sprintf("%s\n", strings.TrimSpace(contributor)))
 	}
+	f.WriteString("```\n")
 
 	f.WriteString(additionalResources)
+
+	g, err := os.Create(r.releaseNotesReadme)
+	if err != nil {
+		return err
+	}
+	defer g.Close()
+
+	g.WriteString(fmt.Sprintf("# %s release notes\n", r.name))
+	g.WriteString("\n")
+	g.WriteString(fmt.Sprintf("The most recent release is [%s](CHANGELOG-%s.md).\n", r.tag, r.tag))
+
+	// TODO list past releases too
```

Comment:
Instead of all past releases maybe list only the supported ones (last three IIRC)?

Reply:
k8s lists [all releases](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/README.md), which is nice considering that GitHub uses alphabetical sorting when listing files and so you get 1.10 listed before 1.2, which itself is listed before 1.20.

Either way, I'm not going to tackle that part right now so deciding how many releases to list is left as an exercise for the reader ;)

Reply:
ACK